omniduct/databases/hiveserver2.py

```diff
@@ -184,75 +318,91 @@ def _run_in_hivecli(self, cmd):
-        col_sanitized = re.sub('\W', '', col.lower().replace(' ', '_'))
-        hive_type = DTYPE_KIND_HIVE_TYPE[dtype.kind]
-        columns.append('  {column}  {type}'.format(column=col_sanitized,
-                                                   type=hive_type))
-
-    cmd = Template("""
-    {% if drop %}
-    DROP TABLE IF EXISTS {{ schema }}.{{ table }};
-    {% endif -%}
-    CREATE TABLE IF NOT EXISTS {{ schema }}.{{ table }} (
-    {%- for col in columns %}
-     {{ col }} {% if not loop.last %}, {% endif %}
-    {%- endfor %}
-    )
-    {%- if text %}
-    ROW FORMAT DELIMITED
-    FIELDS TERMINATED BY "{{ sep }}"
-    STORED AS TEXTFILE
-    {% endif %}
-    {%- if loc %}
-    LOCATION "{{ loc }}"
-    {%- endif %}
-    ;
-    """).render(drop=drop, table=table, schema=schema, columns=columns, text=text, sep=sep)
-
-    logger.debug('Create Table Statement: {}'.format(cmd))
-    return cmd
+    @classmethod
+    def _create_table_statement_from_df(cls, df, table, schema='default', drop=False,
+                                        text=True, sep=None, loc=None,
+                                        table_props=None, partition_cols=None,
+                                        dtype_overrides=None):
+        """
+        Return create table statement for new hive table based on pandas dataframe.
+
+        Parameters:
+            df (pandas.DataFrame, pandas.Series): Used to determine column names
+                and types for create table statement.
+            table (str): The name of the target table.
+            schema (str): The name of the target schema.
+            drop (bool): Whether to include a drop table statement before the
+                create table statement.
+            text (bool): Whether data will be stored as a textfile.
+            sep (str): The separator used by the text data store (defaults to
+                CTRL-A).
+            loc (str): Desired HDFS location (if not the default).
+            table_props (dict): The table properties (if any) to set on the table.
+            partition_cols (list): The columns by which the created table should
+                be partitioned.
+
+        Returns:
+            str: The Hive SQL required to create the table with the above
+                configuration.
+        """
+        table_props = table_props or {}
+        partition_cols = partition_cols or []
+        dtype_overrides = dtype_overrides or {}
+
+        # dtype kind to hive type mapping dict.
+        DTYPE_KIND_HIVE_TYPE = {
+            'b': 'BOOLEAN',  # boolean
+            'i': 'BIGINT',   # signed integer
+            'u': 'BIGINT',   # unsigned integer
+            'f': 'DOUBLE',   # floating-point
+            'c': 'STRING',   # complex floating-point
+            'O': 'STRING',   # object
+            'S': 'STRING',   # (byte-)string
+            'U': 'STRING',   # Unicode
+            'V': 'STRING'    # void
+        }
+        sep = sep or chr(1)
```

Comment:
can you comment on why this character? probably should be moved up to the other defaults (above the DTYPE map)

Reply:
`chr(1)` is the CTRL-A character, which is the default separator used by Hive when tables are stored as text files. It's a way better choice than `r'\t'`, since the tab character is not at all infrequent in strings that one might want to store.